src/components/LinterProvider/index.tsx

```diff
@@ -113,6 +115,8 @@ const mapStateToProps = (
   };
 };
 
-export default connect(mapStateToProps)(
-  LinterProviderBase,
-) as React.ComponentType<PublicProps & Partial<DefaultProps>>;
```

Comment:
I removed this type hint and tested it by rendering with unknown props. I got the expected type error. I'm not sure why but... 🎉

Reply:
Interesting. I just added something like this to a recent patch at your request. Is it somehow specific to the component?

Reply:
Well, previously, the loss of type safety only affected `connect()`. For `withRouter`, it worked unless you were wrapping `connect()`, of course.